runtime/src/bank.rs

```diff
@@ -3221,11 +3228,8 @@ impl Bank {
 
     // gating this under #[cfg(feature = "dev-context-only-utils")] isn't easy due to
     // solana-program-test's usage...
-    pub fn register_unique_recent_blockhash_for_test(&self) {
-        self.register_recent_blockhash(
-            &Hash::new_unique(),
-            &BankWithScheduler::no_scheduler_available(),
-        )
+    pub fn register_unique_recent_blockhash_for_test(&self, hash: &Hash) {
```

Comment:
This new fn signature feels weird, because the method isn't registering a unique hash anymore. We could rename this, but I would lean toward adding a new `register_recent_blockhash_for_test` method instead. That way we can be more explicit about where the new functionality is used.